dwave/optimization/model.py

```diff
@@ -40,7 +40,61 @@
 
     _ShapeLike: typing.TypeAlias = typing.Union[int, collections.abc.Sequence[int]]
 
-__all__ = ["Model"]
+__all__ = ["Expression", "Model"]
+
+
+class Expression(_Graph):
+    def __init__(
+        self,
+        num_inputs: int = 0,
+        lower_bound: typing.Optional[float] = None,
+        upper_bound: typing.Optional[float] = None,
+        integral: typing.Optional[bool] = None,
+    ):
+        self.output: typing.Optional[ArraySymbol] = None
+
+        if num_inputs > 0:
+            if (lower_bound is None or upper_bound is None or integral is None):
+                raise ValueError(
+                    "`lower_bound`, `upper_bound` and `integral` must be provided "
+                    "explicitly when initializing inputs"
+                )
+            for _ in range(num_inputs):
+                self.input(lower_bound, upper_bound, integral)
+
+    def input(self, lower_bound: float, upper_bound: float, integral: bool):
+        """TODO"""
+        # avoid circular import
+        from dwave.optimization.symbols import Input
+        # Shape is always scalar for now
+        return Input(self, lower_bound, upper_bound, integral, shape=tuple())
+
+    def set_output(self, value: ArraySymbol):
+        """TODO"""
+        self.output = value
```

Comment:
Probably we want to set this as the objective for the `_Graph`? So `super().minimize(value)`. Even better:
1) Rename `_Graph.minimize()` -> `_Graph._set_objective()`
2) Add `Model.minimize()` wrapping it
3) Call it from here.

Reply:
Ended up doing exactly this.